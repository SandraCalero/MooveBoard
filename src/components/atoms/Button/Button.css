/* Button base
   ============================= */

.button {
	border: none;
	margin: var(--margin-regular);
	padding: var(--no-padding);
	background: none;
	color: inherit;
}

.button img {
	width: var(--image-size);
	height: var(--image-size);
	margin: var(--margin-regular);
}

.button:hover,
.button img:hover {
	opacity: var(--hover-opacity);
}

.button:active {
	transform: var(--click-button-transform);
}

/* New Post It Button */

.button.newPostItButton {
	z-index: 4;
}

/* Modal buttons */

.button.confirmButton,
.button.cancelButton {
	width: 10rem;
	height: 5.2rem;
	border-radius: var(--border-radius-regular);
}

.button.confirmButton {
	background: var(--blue);
	color: var(--white);
}

.button.cancelButton {
	background: var(--light-gray);
}

/* Trash space buttons */

.button.emptyTrash,
.button.restorePostIts {
	display: flex;
	flex-direction: column;
	align-items: center;
}

/* Post It button */

.button.closePostIt {
	align-self: flex-end;
	opacity: 0.2;
	font-size: 2rem;
	padding: 0.8rem 1.6rem 0;
	margin: 0;
}

.button.closePostIt:hover {
	opacity: 1;
}
