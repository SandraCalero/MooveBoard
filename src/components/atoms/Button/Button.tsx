import './Button.css';

interface ButtonProps {
	text?: string;
	icon?: string;
	variant: string;
	altText?: string;
	onClick: () => void;
}

export default function Button({
	text,
	variant,
	icon,
	altText,
	onClick,
}: ButtonProps) {
	return (
		<button
			name='button'
			type='button'
			className={`flex flex-col items-center border-none m-2 p-0 bg-transparent hover:opacity-90 active:translate-y-1 ${variant}`}
			onClick={onClick}
		>
			{icon && <img className='h-4/5 w-4/5' src={icon} alt={altText} />}
			{text}
		</button>
	);
}

Button.defaultProps = {
	text: '',
	icon: '',
	altText: '',
};
