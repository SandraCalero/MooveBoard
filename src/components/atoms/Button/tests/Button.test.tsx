import { render } from 'utils/tests/test-utils';
import Button from '../Button';

describe('<Button />', () => {
	it('Snapshot', () => {
		const onClick = jest.fn();
		const { container, getByText } = render(
			<Button
				variant='justify-center w-24 h-12 rounded-lg bg-grey
						dark:bg-mine-shaft-500 dark:text-white transition-colors'
				onClick={onClick}
				text='Cancel'
			/>
		);
		expect(container).toMatchSnapshot();
		expect(getByText('Cancel')).toBeTruthy();
	});
});
