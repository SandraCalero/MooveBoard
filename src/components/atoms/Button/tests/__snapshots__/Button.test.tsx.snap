// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
<div>
  <button
    class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 justify-center w-24 h-12 rounded-lg bg-grey dark:bg-mine-shaft-500 dark:text-white transition-colors"
    name="button"
    type="button"
  >
    Cancel
  </button>
</div>
`;
